From 2ece4eac71aec062ebf6d20610028c3ecd007085 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:47 -0400
Subject: [tools] perf/build: Fix non-existent build directory handling

Message-id: <1408462094-14194-179-git-send-email-jolsa@redhat.com>
Patchwork-id: 88167
O-Subject: [PATCH RHEL7.1 BZ1131394 178/325] tools/perf/build: Fix non-existent build directory handling
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 3fb66335e13ef7426affe9efa48c08857202c1cb
Author: Ingo Molnar <mingo@kernel.org>
Date: Wed Oct 9 17:00:23 2013 +0200

description
===========
Arnaldo reported that non-existent build directories were not
recognized  properly. The reason is readlink failure causing 'O'
to become empty.

Solve it by passing through the 'O' variable unmodified if
readlink fails.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/Makefile b/tools/perf/Makefile
index 5aa3d04..9147044 100644
--- a/tools/perf/Makefile
+++ b/tools/perf/Makefile
@@ -34,7 +34,7 @@ endif
 # Only pass canonical directory names as the output directory:
 #
 ifneq ($(O),)
-  FULL_O := $(shell readlink -f $(O))
+  FULL_O := $(shell readlink -f $(O) || echo $(O))
 endif
 
 define print_msg
-- 
1.7.1